### DISCUSSION TITLE
Additional information on the route from point A to B

### DISCUSSION THREAD

**Discussion by Marskiner** [organicmaps/organicmaps#746] [Ideas]
Hi guys,
Do you think it will be possible to have something like this in OM? A brief description of the directions when you want to have an idea about the path from the point A to B. Currently, you only know the KM and required time but for me, it's useful to have the directions as well.
Is it too complex to implement?
I've seen it in Magic Earth, Osmand+ etc.
![Road-dir-Graphhopper](https://user-images.githubusercontent.com/81854689/123511104-c1782800-d67f-11eb-8336-73be4d9bdc18.png)
![Road-dir-OSRM](https://user-images.githubusercontent.com/81854689/123511111-cb019000-d67f-11eb-91d4-70ecda7f98c1.png)

**Comment by biodranik** [comment]
Please describe in more detail why exact directions are important for you and how exactly you're using them to plan the trip? It's clear that when you drive, they may be useful in a printed way or on the screen, but not very clear while planning.

**Reply by Marskiner** [reply]
The directions as you can see highlighted in red in the two photos above (e.g., turn left onto Marylebone Road (A501) 900m etc. In my opinion, they are useful when you are planning ahead a trip and definitely need details. That's why Google and Apple Maps both have them. Sometimes you have very few directions to follow, so why do I need to turn on the GPS, be in that city, activate my GEO location when in a few seconds you can see directions as if you had a paper map? Got it?
In OM, you MUST be in that location to get directions to reach the destination. You can't get a preview of the route directions if you are not physically there.
Imagine you have problem with your GPS and/or your mobile network connection is down.  How do I get to that location in OM? How can I get directions? You can't.
Instead, if you had route directions, you could follow all the steps (turn left onto..,turn right onto etc.) and reach your final destination as if you were following a paper map.

**Reply by biodranik** [reply]
Thanks for explaining it. OM works without an internet connection, but without GPS, it can't help. You're right.
My use case is when GPS is lost: in tunnels, for example. But at that time, you're already on the road.

My question was about planning because I can't fully understand the use case and its value. If you plan hours or days before the trip, you are not likely to remember directions. If you plan just now, before starting the trip, it may be useful when you don't want to use navigation and remember the nearest turn. But it looks to me like an unusual case because most people check the visual route first, including length, ETA, and the closest turn/road.
